tools/monorepo-utils/src/core/github/repo.ts

```diff
@@ -167,3 +170,51 @@ export const createPullRequest = async ( options: {
 	//@ts-ignore There is a type mismatch between the graphql schema and the response. pullRequest.data.head.repo.has_discussions is a boolean, but the graphql schema doesn't have that field.
 	return pullRequest.data;
 };
+
+/**
+ * Get a pull request from Github.
+ *
+ * @param {Object} options
+ * @param {string} options.owner repository owner.
+ * @param {string} options.name  repository name.
+ * @param          prNumber      pull request number.
+ * @return {Promise<object>}     pull request data.
+ */
+export const getPullRequest = async (
+	options: { owner: string; name: string },
+	prNumber: string
+): Promise< PullRequestEndpointResponse[ 'data' ] > => {
+	const { owner, name } = options;
+	const pr = await octokitWithAuth().request(
+		'GET /repos/{owner}/{repo}/pulls/{pull_number}',
+		{
+			owner,
+			repo: name,
+			pull_number: Number( prNumber ),
+		}
+	);
+
+	//@ts-ignore
+	return pr.data;
+};
+
+/**
+ * Determine if a pull request is coming from a community contribution. ie, not from a member of the woocommerce org.
+ *
+ * @param {Object} pullRequestData pull request data.
+ * @param {string} owner           repository owner.
+ * @param {string} name            repository name.
+ * @return {boolean} if a pull request is coming from a community contribution.
+ */
+export const isCommunityPullRequest = (
+	pullRequestData: PullRequestEndpointResponse[ 'data' ],
```

Comment:
You have a good point. The type would look something like this:
```
{
  author_association: string,
  head: {
    repo: {
      full_name: string,
    }
  }
}
```
For me thats inferring a lot of assumptions about the shape of the data. I'd prefer to use the one from Octokit because its less error prone.